server/services/utils/helpers.js

```diff
@@ -26,3 +26,38 @@ export const isIndexNotFoundError = (err) => {
       'Configured indices are not found: [.opendistro-alerting-config]'
   );
 };
+
+export function getClientBasedOnDataSource(
+  context,
+  dataSourceEnabled,
+  dataSourceId,
+  request,
+  client
+) {
+  if (dataSourceEnabled && dataSourceId && dataSourceId.trim().length !== 0) {
+    // Client for remote cluster
+    return context.dataSource.opensearch.legacy.getClient(dataSourceId).callAPI;
+  } else {
+    // Fall back to default local cluster
+    return client.asScoped(request).callAsCurrentUser;
+  }
+}
+
+export function validateQuery(dataSourceEnabled, fields = {}) {
+  let query = schema.object({}, { unknowns: 'allow' });
```

Comment:
Why allow unknows?

Reply:
also, let create the schema first and in the end use it to create the return value;
It will simplify the code

```
const schemaObj = {
    ...fields
};

if (dataSourceEnabled) {
    schemaObj['dataSourceId'] = ...
}

return schema.object(schemaObj)
```

Reply:
addressed